ownCloud/Client/Actions/Actions+Extensions/DuplicateAction.swift

```diff
@@ -34,37 +37,126 @@ class DuplicateAction : Action {
 
 	// MARK: - Action implementation
 	override func run() {
-		guard context.items.count > 0, let core = self.core, let item = context.items.first, let itemName = item.name else {
+		guard context.items.count > 0 else {
 			completed(with: NSError(ocError: .itemNotFound))
 			return
 		}
+		remainingItems.append(contentsOf: context.items)
+		proceed()
+	}
 
-		context.items.forEach({ (item) in
-			if let rootItem = item.parentItem(from: core) {
+	func proceed() {
+		guard let item = remainingItems.first, let baseName = item.baseName, let core = self.core, let itemPath = item.path, let rootItemPath = itemPath.parentPath() else {
+			completed(with: NSError(ocError: .itemNotFound))
+			return
+		}
 
-				var name: String = "\(itemName) copy"
+		let pattern = " \(localizedCopy)[ ]?[0-9]{0,}$"
+		let searchName = baseName.replacingOccurrences(for: pattern)
 
-				if let itemFileExtension = item.fileExtension, let baseName = item.baseName {
-					var fileExtension = itemFileExtension
+		query = OCQuery(forPath: rootItemPath)
+		query?.delegate = self
+		query?.queryItem = item
 
-					if fileExtension != "" {
-						fileExtension = ".\(fileExtension)"
-					}
+		let filterHandler: OCQueryFilterHandler = { (_, _, item) -> Bool in
```

Comment:
@hosy I think for all this filtering stuff, one or two lines of comments wouldn't hurt

Reply:
In my opinion it is not so difficult to understand. It is only a check if a baseName has a prefix.